src/IntermediateRepresentation.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * IntermediateRepresentation.js - representation of the results of parsing
+ * an input file 
+ *
+ * Copyright © 2023 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @class Representation of parser results
+ * @abstract
```

Comment:
Perhaps instead of an abstract class, it could be concrete but generic `IntermediateRepresentation<T>`?